src/usr/local/www/services_dhcp.php

```diff
@@ -581,7 +591,15 @@ function validate_partial_mac_list($maclist) {
 		$dhcpdconf['domain'] = $_POST['domain'];
 		$dhcpdconf['domainsearchlist'] = $_POST['domainsearchlist'];
 		$dhcpdconf['ignorebootp'] = ($_POST['ignorebootp']) ? true : false;
-		$dhcpdconf['denyunknown'] = ($_POST['denyunknown']) ? true : false;
+
+		if ($_POST['denyunknown'] == "enabled") {
+			$dhcpdconf['denyunknown'] = "enabled";
+		} else if ($_POST['denyunknown'] == "class") {
+			$dhcpdconf['denyunknown'] = "class";
+		} else if (isset($_POST['denyunknown'])) {
```

Comment:
Why test that last case? It would be better to just unset it unconditionally if the first two cases fail.

Or consider rewriting it to a more adaptable method with less duplication of text:

```
if (in_array($_POST['denyunknown'], array("enabled", "class"))) {
	$dhcpdconf['denyunknown'] = $_POST['denyunknown'];
} else {
	unset($dhcpdconf['denyunknown']);
}
```